mattermost.go

```diff
@@ -6,39 +6,339 @@ package main
+var flagConfigFile string
+var flagEmail string
+var flagPassword string
+var flagTeamName string
+var flagRole string
+var flagRunCmds bool
+
 func main() {
 
-	pwd, _ := os.Getwd()
-	fmt.Println("Current working directory is set to " + pwd)
+	parseCmds()
 
-	var config = flag.String("config", "config.json", "path to config file")
-	flag.Parse()
+	utils.LoadConfig(flagConfigFile)
+
+	if flagRunCmds {
+		utils.ConfigureCmdLineLog()
+	}
+
+	pwd, _ := os.Getwd()
+	l4g.Info("Current working directory is %v", pwd)
+	l4g.Info("Loaded config file from %v", utils.FindConfigFile(flagConfigFile))
 
-	utils.LoadConfig(*config)
 	api.NewServer()
 	api.InitApi()
 	web.InitWeb()
-	api.StartServer()
 
-	// If we allow testing then listen for manual testing URL hits
-	if utils.Cfg.ServiceSettings.AllowTesting {
-		manualtesting.InitManualTesting()
+	if flagRunCmds {
+		runCmds()
+	} else {
+		api.StartServer()
+
+		// If we allow testing then listen for manual testing URL hits
+		if utils.Cfg.ServiceSettings.AllowTesting {
+			manualtesting.InitManualTesting()
+		}
+
+		// wait for kill signal before attempting to gracefully shutdown
+		// the running service
+		c := make(chan os.Signal)
+		signal.Notify(c, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+		<-c
+
+		api.StopServer()
+	}
+}
+
+func parseCmds() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, usage)
 	}
 
-	// wait for kill signal before attempting to gracefully shutdown
-	// the running service
-	c := make(chan os.Signal)
-	signal.Notify(c, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
-	<-c
+	flag.StringVar(&flagConfigFile, "config", "config.json", "")
+	flag.StringVar(&flagEmail, "email", "", "")
+	flag.StringVar(&flagPassword, "password", "", "")
+	flag.StringVar(&flagTeamName, "team_name", "", "")
+	flag.StringVar(&flagRole, "role", "", "")
 
-	api.StopServer()
+	flag.BoolVar(&flagCmdCreateTeam, "create_team", false, "")
+	flag.BoolVar(&flagCmdCreateUser, "create_user", false, "")
+	flag.BoolVar(&flagCmdAssignRole, "assign_role", false, "")
+	flag.BoolVar(&flagCmdResetPassword, "reset_password", false, "")
+
+	flag.Parse()
+
+	flagRunCmds = flagCmdCreateTeam || flagCmdCreateUser || flagCmdAssignRole || flagCmdResetPassword
+}
+
+func runCmds() {
+	cmdCreateTeam()
+	cmdCreateUser()
+	cmdAssignRole()
+	cmdResetPassword()
+}
+
+func cmdCreateTeam() {
+	if flagCmdCreateTeam {
+		if len(flagTeamName) == 0 {
+			fmt.Fprintln(os.Stderr, "flag needs an argument: -team_name")
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		if len(flagEmail) == 0 {
+			fmt.Fprintln(os.Stderr, "flag needs an argument: -email")
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		c := &api.Context{}
+		c.RequestId = model.NewId()
+		c.IpAddress = "cmd_line"
+
+		team := &model.Team{}
+		team.DisplayName = flagTeamName
+		team.Name = flagTeamName
+		team.Email = flagEmail
+		team.Type = model.TEAM_INVITE
+
+		api.CreateTeam(c, team)
+		if c.Err != nil {
+			if c.Err.Message != "A team with that domain already exists" {
```

Comment:
It was intentional so someone could script this and not worry about this failure since their intent was satisfied.